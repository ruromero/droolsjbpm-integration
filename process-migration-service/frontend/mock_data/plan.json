{
  "id": 99,
  "name": "mockupTest1",
  "description": "mockup_",
  "mappings": {
    "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E":
      "_B8E4DA1E-A62B-49C2-9A94-FEE5F5FD2B4E"
  },
  "source": {
    "containerId": "evaluation_2.0.0-SNAPSHOT",
    "processId": "old_evaluation"
  },
  "target": {
    "containerId": "evaluation_3.0.0-SNAPSHOT",
    "processId": "new_evaluation"
  }
}