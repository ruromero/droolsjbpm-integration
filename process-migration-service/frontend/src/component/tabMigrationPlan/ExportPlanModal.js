import React from "react";

import { Button } from "patternfly-react";
import { Modal } from "patternfly-react";
import { Icon } from "patternfly-react";
import { OverlayTrigger } from "patternfly-react";
import { Tooltip } from "patternfly-react";
import {
  Form,
  ControlLabel,
  FormGroup,
  FormControl
} from "patternfly-react/dist/js/components/Form";
export default class ExportPlanModal extends React.Component {
  constructor() {
    super();
    this.state = { showModal: false };
  }
  closeModal = () => {
    this.setState({ showModal: false });
  };
  openModal = () => {
    this.setState({ showModal: true });
  };

  copyToClipboard = () => {
    navigator.clipboard.writeText(this.planToString());
    this.setState({ showModal: false });
  };

  planToString = () => {
    return JSON.stringify(
      this.props.plan,
      (key, value) => {
        if (key !== "id" && value !== null) {
          return value;
        }
      },
      2
    );
  };

  render() {
    const tooltipExport = (
      <Tooltip id="tooltip">
        <div>Export migration plan</div>
      </Tooltip>
    );

    const formBody = (
      <Form>
        <FormGroup controlId="exportText">
          <ControlLabel>Migration plan</ControlLabel>
          <FormControl
            componentClass="textarea"
            value={this.planToString()}
            rows={10}
            readOnly
          />
        </FormGroup>
      </Form>
    );

    return (
      <React.Fragment>
        <OverlayTrigger overlay={tooltipExport} placement={"bottom"}>
          <Button bsStyle="link" onClick={this.openModal}>
            <Icon type="pf" name="export" />
          </Button>
        </OverlayTrigger>

        <Modal show={this.state.showModal} onHide={this.closeModal} size="lg">
          <Modal.Header>
            <Modal.CloseButton onClick={this.closeModal} />
            <Modal.Title>Export migration plan</Modal.Title>
          </Modal.Header>
          <Modal.Body>{formBody}</Modal.Body>
          <Modal.Footer>
            <Button
              bsStyle="default"
              className="btn-cancel"
              onClick={this.closeModal}
            >
              Cancel
            </Button>
            <Button bsStyle="primary" onClick={this.copyToClipboard}>
              Copy to clipboard
            </Button>
          </Modal.Footer>
        </Modal>
      </React.Fragment>
    );
  }
}
