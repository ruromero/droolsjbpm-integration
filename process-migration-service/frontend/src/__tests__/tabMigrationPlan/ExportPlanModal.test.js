import renderer from "react-test-renderer";
import React from "react";
import ExportPlanModal from "../../component/tabMigrationPlan/ExportPlanModal";
import plan from "../../../mock_data/plan.json";

test("ExportPlanModal renders correctly using snapshot", () => {
  const tree = renderer.create(<ExportPlanModal plan={plan} />).toJSON();
  expect(tree).toMatchSnapshot();
});
