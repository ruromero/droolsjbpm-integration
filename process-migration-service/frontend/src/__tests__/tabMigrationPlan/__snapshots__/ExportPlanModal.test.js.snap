// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportPlanModal renders correctly using snapshot 1`] = `
<button
  className="btn btn-link"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
  type="button"
>
  <span
    aria-hidden="true"
    className="pficon pficon-export"
  />
</button>
`;
